### ISSUE TITLE
Allow B button to exit tests that are being run in auto mode

### ISSUE THREAD

**Issue by abaire** [abaire/nxdk_pgraph_tests#29]
Currently it's impossible to break out if the tester goes into auto run mode (either via timeout on the main menu or pressing 'X'). Ideally users should be able to break out of auto run mode and get back to the menu by pressing 'B' at any time.

**Comment by akarsh786** [comment]
hey can i work on this issue

**Comment by abaire** [comment]
> hey can i work on this issue

Definitely. As a pointer you'll want to look at making [`TestSuite::RunAll`](https://github.com/abaire/nxdk_pgraph_tests/blob/829c658bc2fbcaa01996a84076e04bebf036e3e0/tests/test_suite.h#L22) substantially smarter. 

I'd probably either return execution to the [`TestDriver`'s run loop](https://github.com/abaire/nxdk_pgraph_tests/blob/5b4b2a6db8dfc90fb2905c8f408ab0215e76131e/test_driver.cpp#L26) so it can check for input or move the input event loop handling to a background thread that updates some shared state that the TestSuite can check in between tests.
